polynote-kernel/src/main/scala/polynote/kernel/interpreter/scal/SemanticDbScan.scala

```diff
@@ -0,0 +1,89 @@
+package polynote.kernel.interpreter
+package scal
+
+import polynote.kernel.{BaseEnv, ScalaCompiler}
+import polynote.kernel.dependency.Artifact
+import polynote.kernel.environment.CurrentTask
+import polynote.kernel.logging.Logging
+import polynote.kernel.task.TaskManager
+import zio.ZIO.effectTotal
+import zio.blocking.{Blocking, effectBlocking}
+import zio.clock.Clock
+import zio.duration.Duration
+import zio.{RIO, Ref, Task, ZIO}
+
+import scala.meta.interactive.InteractiveSemanticdb
+import scala.reflect.internal.util.{BatchSourceFile, SourceFile}
+import scala.reflect.io.{AbstractFile, FileZipArchive}
+import scala.tools.nsc.{Global, interactive}
+import scala.tools.nsc.interactive.{NscThief, Response}
+
+class SemanticDbScan (compiler: ScalaCompiler) {
+
+  private val classpath = compiler.global.classPath.asClassPathString//.dependencies.map(_.file.getAbsolutePath).mkString(File.pathSeparator)
+  private val sources = compiler.dependencies.flatMap(_.source)
+  val semanticdbGlobal: interactive.Global = InteractiveSemanticdb.newCompiler(classpath, List())
+  private val importer = semanticdbGlobal.mkImporter(compiler.global)
+
+  def init: RIO[BaseEnv with TaskManager, Unit] = TaskManager.run("semanticdb", "Scanning sources")(scanSources).flatMap {
+    sources => TaskManager.run("semanticdb", "Indexing sources")(indexSources(sources))
```

Comment:
How long does this sort of thing tend to take with the typical dependencies we have?